src/ADService.js

```diff
@@ -55,16 +55,33 @@ class ADService {
     new Date().getTime() < tokenResult.expiresOn * 1000;
 
   getAccessTokenAsync = async () => {
-    if (!this._isTokenValid(this.tokenResult)) {
-      const result = await this.fetchAndSetTokenAsync(
-        this.tokenResult.refreshToken,
-        this.loginPolicy,
-        true
-      );
-
-      if (!result.isValid) {
-        return result;
+    try {
+      // Additional is valid to store result from second request
+      let isValid = false;
+      if (!this._isTokenValid(this.tokenResult)) {
+        const result = await this.fetchAndSetTokenAsync(
```

Comment:
Consider renaming `result` and `result2` to something more informative to indicate the differences between the responses. E.g. `resultSignin` and `resultReset`